daemon/daemon.go

```diff
@@ -1462,6 +1462,23 @@ func mapValidateWalker(path string) error {
 	return nil
 }
 
+// TriggerReload reloads the BPF datapath for the base programs and all
+// endpoints without recalculating policy.
+//
+// If an error is returned, then no regeneration was successful. If no error
+// is returned, then the base programs were successfully regenerated, but
+// endpoints may or may not have successfully regenerated.
+//
+// The returned WaitGroup allows the caller to wait until all endpoints have
+// finished regenerating.
+func (d *Daemon) TriggerReload(context string) (*sync.WaitGroup, error) {
```

Comment:
Can you make the naming here more explicit? This really only compiles the base programs and *reloads* the endpoint programs without recompiling?

Reply:
That's the intention, but it's not optimized to skip the compile yet (I remember @ianvernon briefly discussing adding that "skip compile" piece as one of his PRs, but I'll sync with him on the status).

I can rename this to `TriggerReloadWithoutRecompile()`, perhaps?

Reply:
Do you have a recommendation for a more explicit name? Right now I'm leaning more towards something like `ReloadDatapath()` or `ReloadBPF()` because the point of this function is that it requires Reload, but not necessarily compile/policy recalculation/etc. Whether or not the underlying code is efficient enough to avoid compile/policy recalculation/etc is a matter for the implementation, not for the callers.

(I'll update the comment, too)

Reply:
I believe that we only need this patch to achieve the above optimization of skipping compile from this path, on top of the existing PR here:

https://github.com/joestringer/cilium/commit/f07ce4e2bbc24f8c37a795e2401d74a0b5af8d6c

But I'd rather split it into its own PR for closer review.

Reply:
I would appreciate the `WithoutRecompile` personally but I'm also fine with crystal clear function doc

Reply:
On one hand, I could see how the `WithoutCompile` part could get out of sync with the underlying implementation.

On the other hand, I can also see how for future (potential) users of the function, it would be important to have that part in the name, to determine if it's a safe version to use for triggering regeneration. 

I can add that in after I see another green build (so nothing broke during rebuild).